src/Servers/IIS/IIS/src/BadHttpRequestException.cs

```diff
@@ -1,23 +1,22 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.IO;
 using System.Runtime.CompilerServices;
 using Microsoft.AspNetCore.Http;
 
 namespace Microsoft.AspNetCore.Server.IIS
 {
-    public sealed class BadHttpRequestException : IOException
+    [Obsolete("Moved to Microsoft.AspNetCore.Http.BadHttpRequestException")]
+    public sealed class BadHttpRequestException : Microsoft.AspNetCore.Http.BadHttpRequestException
     {
         private BadHttpRequestException(string message, int statusCode, RequestRejectionReason reason)
-            : base(message)
+            : base(message, statusCode)
         {
-            StatusCode = statusCode;
             Reason = reason;
         }
 
-        public int StatusCode { get; }
```

Comment:
How concerned are we about the binary breaking change here? It could be mitigated with:
`public new int StatusCode { get => base.StatusCode }`